stock_inventory_valuation_report/reports/stock_inventory_valuation_report.py

```diff
@@ -0,0 +1,120 @@
+# Copyright 2019 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.osv import expression
+
+
+class StockInventoryValuationView(models.TransientModel):
+    _name = "stock.inventory.valuation.view"
+    _description = "Stock Inventory Valuation View"
+
+    report_id = fields.Many2one("report.stock.inventory.valuation.report")
+
+    name = fields.Char()
+    reference = fields.Char()
+    barcode = fields.Char()
+    qty_at_date = fields.Float()
+    uom_id = fields.Many2one(
+        comodel_name="uom.uom",
+    )
+    currency_id = fields.Many2one(
+        comodel_name="res.currency",
+    )
+    cost_currency_id = fields.Many2one(
+        comodel_name="res.currency",
+    )
+    standard_price = fields.Float()
+    stock_value = fields.Float()
+    cost_method = fields.Char()
+
+
+class StockInventoryValuationReport(models.TransientModel):
+    _name = "report.stock.inventory.valuation.report"
+    _description = "Stock Inventory Valuation Report"
+
+    # Filters fields, used for data computation
+    company_id = fields.Many2one(
+        comodel_name="res.company",
+    )
+    inventory_datetime = fields.Datetime(required=True, default=fields.Datetime.now)
+
+    results = fields.One2many(
+        comodel_name="stock.inventory.valuation.view",
+        inverse_name="report_id",
+        compute="_compute_results",
+    )
+
+    @api.depends("inventory_datetime")
+    def _compute_results(self):
+        self.ensure_one()
+        domain = [("type", "=", "product")]
+        product_id = self.env.context.get("product_id", False)
```

Comment:
By default `get` function returned `None` value, because in `if` is `None` equal to `False`.